nebula-unity/Assets/scripts/DungeonMapGenerator.cs

```diff
@@ -209,6 +225,28 @@ public class PotentialDoor {
         return mapScript;
     }
 
+    int[] ReplaceSentinels(int xSize, int ySize, int minDiam, int maxDiam, int maxatt){
+        int[] sentsreturn = new int[5];
+        if (xSize == -1)
+            sentsreturn[0] = XDIMDEFAULT;
+        if (ySize == -1)
+            sentsreturn[1] = YDIMDEFAULT;
+        if (minDiam == -1)
+            sentsreturn[2] = minRoomDiameter;
+        if (maxDiam == -1)
+            sentsreturn[3] = maxRoomDiameter;
+        if (maxatt == -1)
+            sentsreturn[4] = maxAttempts;
+        return sentsreturn;
+    }
+
+    float ReplaceSentinelsFloat(float sfact){
+        if (sfact == -1f)
+            return shapeFactor;
+        else
+            return sfact;
+    }
+
```

Comment:
Your problem likely lies here.  ReplaceSentinelsFloat() looks good, it replaces the sentinel value if it exists, otherwise it returns the original value.    However let's take a look at ReplaceSentinels().  If `maxatt != -1`, what happens?  What does maxatt get set to on line 135?  What should it be?

Reply:
Msg me once you figure it out but before you make the change, as I have a good solution for this.